Library/Homebrew/extend/io.rb

```diff
@@ -1,22 +1,24 @@
-# typed: false
+# typed: true
 # frozen_string_literal: true
 
 class IO
   def readline_nonblock(sep = $INPUT_RECORD_SEPARATOR)
     line = +""
     buffer = +""
 
-    loop do
-      break if buffer == sep
+    begin
+      loop do
+        break if buffer == sep
 
-      read_nonblock(1, buffer)
-      line.concat(buffer)
-    end
+        read_nonblock(1, buffer)
+        line.concat(buffer)
+      end
 
-    line.freeze
-  rescue IO::WaitReadable, EOFError => e
-    raise e if line.empty?
+      line.freeze
+    rescue IO::WaitReadable, EOFError
+      raise if line.empty?
```

Comment:
`IO::WaitReadable` is not technically an Exception, so it can't be passed to `raise`, but we can avoid the issue using an empty re-raise instead (`e` is not needed otherwise)